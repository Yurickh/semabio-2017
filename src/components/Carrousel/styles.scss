@import 'ambience/color';
@import 'base/responsive-prop';

.carrousel {
  position: relative;

  & > .fog-before {
    position: absolute;
    z-index: 1;
    top: 0;
    left: 0;

    user-select: none;

    background: linear-gradient(to right, $color-background 70%, rgba(#FFF, 0));

    @include responsive-prop(width, 20, 250);
    @include responsive-prop(height, 200, 200);
  }

  & > .fog-after {
    position: absolute;
    z-index: 1;
    top: 0;
    right: 0;

    user-select: none;

    background: linear-gradient(to left, $color-background 70%, rgba(#FFF, 0));

    @include responsive-prop(width, 20, 250);
    @include responsive-prop(height, 200, 200);
  }

  & > .people {
    position: relative;

    overflow: hidden;

    @include responsive-prop(height, 130, 130);
  }

  & > .people > .person {
    position: absolute;
    left: 50%;

    transition: top 1s, transform .15s ease-in-out;

    @include responsive-prop(top, 15, 15);
  }

  & > .people > .person.-centered {
    top: 0;
  }

  & > .people > .person > .avatar {
    transition: all 1s;

    border-radius: 50%;

    @include responsive-prop(width, 70, 70);
    @include responsive-prop(height, 70, 70);
  }

  & > .people > .person.-centered > .avatar {
    @include responsive-prop(width, 100, 100);
    @include responsive-prop(height, 100, 100);
  }

  & > .bar {
    display: flex;
    justify-content: center;

    width: 100%;

    background-color: #E8E9ED;

    @include responsive-prop(height, 3, 3);
  }

  & > .bar > .middle-bar {
    background-color: #58C166;

    @include responsive-prop(width, 100, 100);
    @include responsive-prop(height, 3, 3);
  }

  & > .info {
    @include responsive-prop(margin-top, 24, 24);
  }

  & > .info > .name {
    @include responsive-prop(font-size, 18, 18);
  }

  & > .info > .about {
    letter-spacing: 2.4px;

    color: $color-light;

    @include responsive-prop(font-size, 12, 12);
    @include responsive-prop(margin-top, 2, 2);
  }
}
