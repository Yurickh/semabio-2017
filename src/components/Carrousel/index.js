import React, { Component } from 'react'

import profilesData from '../../helpers/profilesData'

import './styles.css'

class Carrousel extends Component {
  people = profilesData()
  state = {
    selected: 8
  }

  isSelected = item => item === this.state.selected

  renderPeople = () => {
    return (
      this.people.map((person, i) => (
        <div
          style={this.styleFor(i)}
          className={`person ${this.isSelected(i) ? '-centered' : ''}`}
          key={person.name + i}
          onClick={() => this.moveToIndex(i)}
        >
          <img
            alt="avatar"
            src={person.photo}
            className='avatar'
          />
        </div>
      ))
    )
  }

  styleFor = (index) => {
    let offset = index - this.state.selected
    const numOfPeople = this.people.length

    if (offset < -numOfPeople/2) {
      offset += numOfPeople
    } else if (offset > numOfPeople/2) {
      offset -= numOfPeople
    }

    return {
      transform: `translateX(calc(${offset * 180}px - 50%))`
    }
  }

  moveToNext = () => {
    const selected = (this.state.selected + 1) % this.people.length
    this.setState({ selected  })
  }

  moveToPrevious = () => {
    const numOfPeople = this.people.length
    const selected = (this.state.selected - 1) % numOfPeople

    this.setState({
      selected: selected < 0 ? selected + numOfPeople : selected
    })
  }

  moveToIndex = (selected) => {
    this.setState({ selected })
  }

  currentPerson = () => {
    return this.people[this.state.selected]
  }

  render() {
    return (
      <div className="carrousel">
        <div className="fog-before" onClick={this.moveToPrevious} />
        <div className="people">
          { this.renderPeople() }
        </div>
        <div className="bar">
          <div className="middle-bar" />
        </div>
        <div className="info">
          <div className="name">
            {this.currentPerson().name}
          </div>
          <div className="about">
            {this.currentPerson().about}
          </div>
        </div>
        <div className="fog-after" onClick={this.moveToNext} />
      </div>
    )
  }
}

export default Carrousel
