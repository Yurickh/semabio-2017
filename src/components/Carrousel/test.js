import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Carrousel from '../Carrousel'

describe('Carrousel', () => {
  let carrousel

  beforeEach(() => {
    carrousel = shallow(<Carrousel deterministic />)
  })

  it('should render correctly', () => {
    expect(toJSON(carrousel)).toMatchSnapshot()
  })
})
