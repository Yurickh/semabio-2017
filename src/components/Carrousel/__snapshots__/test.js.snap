// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carrousel should render correctly 1`] = `
<div
  className="carrousel"
>
  <div
    className="fog-before"
    onClick={[Function]}
  />
  <div
    className="people"
  >
    <div
      className="person -centered"
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(0px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="leticia.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(180px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="rodrigo.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(360px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="ademar.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(540px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="joao.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(720px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="pedro.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(900px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="helena.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(1080px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="ana.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(1260px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="dhara.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(-1260px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="sarah.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(-1080px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="giovana.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(-900px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="cibelle.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(-720px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="mariana.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(-540px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="marcela.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(-360px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="isadora.jpg"
      />
    </div>
    <div
      className="person "
      onClick={[Function]}
      style={
        Object {
          "transform": "translateX(calc(-180px - 50%))",
        }
      }
    >
      <img
        alt="avatar"
        className="avatar"
        src="patryck.jpg"
      />
    </div>
  </div>
  <div
    className="bar"
  >
    <div
      className="middle-bar"
    />
  </div>
  <div
    className="info"
  >
    <div
      className="name"
    >
      Let√≠cia Campos
    </div>
    <div
      className="about"
    >
       
    </div>
  </div>
  <div
    className="fog-after"
    onClick={[Function]}
  />
</div>
`;
